drivers/clock_control/clock_control_nrf.c

```diff
@@ -172,9 +178,78 @@ static void set_on_state(uint32_t *flags)
 	irq_unlock(key);
 }
 
+#if CONFIG_CLOCK_CONTROL_NRF_HF_CALIBRATION
+
+static void calibration_handler(struct k_timer *timer)
+{
+	if (nrfx_clock_hfclk_is_running()) {
+		return;
+	}
+	sys_notify_init_spinwait(&hf_cal_cli.notify);
+	(void)onoff_request(z_nrf_clock_control_get_onoff(CLOCK_CONTROL_NRF_SUBSYS_HF),
+			    &hf_cal_cli);
+	k_sem_take(&cal_sem, K_FOREVER);
```

Comment:
can you block in timer handler context? I don't think so:

> An expiry function that is executed each time the timer expires. The function is executed by the system clock interrupt handler.

Can't we use async API for `onoff` (if it exists) to avoid blocking here? Something like:
1) k_timer triggers handler -> request HF clock and return
2) onoff callback is invoked for HF clock being requested -> `clkstarted_handle` must have been already executed, so we can drop HF clock request

Also, how requesting HFINT works in clock_control, as HFINT must be already running if HFXO is not used? (you must have some HF clock running - HFINT if HFXO is not used)